### PR TITLE
Support `throwIfNoEntry` in StatOptions

### PR DESCRIPTION
**Pull request by mausworks**
This PR solves #781

### CODE REVIEW

File: src/volume.ts [hunk 1]
```diff
@@ -365,10 +365,17 @@ const readdirDefaults: IReaddirOptions = {
...
-// Options for `fs.fstat`, `fs.fstatSync`, `fs.lstat`, `fs.lstatSync`, `fs.stat`, and `fs.statSync`
+// Options for `fs.lstat`, `fs.lstatSync`, `fs.stat`, and `fs.statSync`
 export interface IStatOptions {
   bigint?: boolean;
+  throwIfNoEntry?: boolean;
 }
+
+// Options for `fs.fstat`, fs.fstatSync
+export interface IFStatOptions {
+  bigint?: boolean;
```
Comment (mausworks) [inline src/volume.ts:376]: `fstat` does not support this, so I simply split the options

File: src/volume.ts [hunk 2]
```diff
@@ -1499,50 +1506,78 @@ export class Volume {
...
-  lstatSync(path: PathLike, options: { bigint: true }): Stats<bigint>;
-  lstatSync(path: PathLike, options?: IStatOptions): Stats {
-    return this.lstatBase(pathToFilename(path), getStatOptions(options).bigint as any);
+  lstatSync(path: PathLike, options: { throwIfNoEntry?: true | undefined }): Stats<number>;
+  lstatSync(path: PathLike, options: { bigint: false; throwIfNoEntry?: true | undefined }): Stats<number>;
+  lstatSync(path: PathLike, options: { bigint: true; throwIfNoEntry?: true | undefined }): Stats<bigint>;
+  lstatSync(path: PathLike, options: { throwIfNoEntry: false }): Stats<number> | undefined;
+  lstatSync(path: PathLike, options: { bigint: false; throwIfNoEntry: false }): Stats<number> | undefined;
+  lstatSync(path: PathLike, options: { bigint: true; throwIfNoEntry: false }): Stats<bigint> | undefined;
+  lstatSync(path: PathLike, options?: IStatOptions): Stats | undefined {
```
Comment (streamich) [inline src/volume.ts:1532]: Seems like these type overloads are slowly getting out of control.

**Review by streamich — APPROVED** [review]
LGTM. Thank you!

**Comment by mausworks** [comment]
Fantastic. Thanks for your quick review! :+1:

File: src/volume.ts (hunk 2 above)
Comment (mausworks) [reply src/volume.ts:1532]: Indeed! :sweat_smile: 

Should probably be revised if a future version of NodeJS ships with new options.

**Pull request merged**